### PR TITLE
bugfix: fix GlobalStatus=9 can't be cleared in DB storage mode

### LINKED ISSUE #5418: 1.6.1版，global_table表status=9的记录，发生系统不自动删除的问题
image

### PR DESCRIPTION
**Pull request by slievrly**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

fix GlobalStatus=9 can't be cleared in DB storage mode

### Ⅱ. Does this pull request fix one issue?

fix #5418

### CODE REVIEW

File: server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java [hunk 1]
```diff
@@ -435,17 +435,27 @@ protected void handleRetryCommitting() {
...
+            if (GlobalStatus.AsyncCommitting.equals(committingSession.getStatus())) {
+                try {
+                    core.doGlobalCommit(committingSession, true);
+                } catch (TransactionException ex) {
+                    LOGGER.error("Failed to async committing [{}] {} {}", committingSession.getXid(), ex.getCode(), ex.getMessage(), ex);
+                }
+            } else if (GlobalStatus.Committed.equals(committingSession.getStatus())) {
+                try {
+                    if (committingSession.getBranchSessions().isEmpty()) {
+                        SessionHelper.endCommitted(committingSession, true);
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java:454]: 可能存在并发end的问题

Comment (a364176773) [reply server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java:454]: 这个是异步提交，有些地方是retring committing

**Comment by a364176773** [comment]
因为死锁的修复是靠sql改动和索引改动解决的，我更倾斜去捕获这个死锁异常，提示用户检查表索引

File: server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java (hunk 1 above)
Comment (slievrly) [reply server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java:454]: I understand Committed is a final state. In addition to the concurrent processing between nodes, are there other tasks processing this state on the same node?

---

Committed 我理解是个终态，除了节点间的并发处理，在同一个节点有其他任务处理这个状态嘛？

Comment (slievrly) [reply server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java:454]: ok, I  got.

**Review by a364176773 — APPROVED** [review]
LGTM

**Comment by a364176773** [comment]
我不认为这个是一个bug,这是个优化,对于已结束残留的完成事务先前是通过重启来处理的,只不过现在运行运行中处理了

**Pull request merged**